inference-engine/src/mkldnn_plugin/nodes/mkldnn_topk_node.cpp

```diff
@@ -42,128 +1197,113 @@ bool MKLDNNTopKNode::isSupportedOperation(const std::shared_ptr<const ngraph::No
     return true;
 }
 
-MKLDNNTopKNode::MKLDNNTopKNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng,
-                                     MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+MKLDNNTopKNode::MKLDNNTopKNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(op, eng, cache) {
     std::string errorMessage;
-    if (!isSupportedOperation(op, errorMessage)) {
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "TopK layer with name '" + getName() + "'";
+
+        auto topKOp = ngraph::as_type_ptr<ngraph::op::v1::TopK>(op);
+
+        in_dims = topKOp->get_input_partial_shape(TOPK_DATA);
+        out_dims = topKOp->get_output_partial_shape(TOPK_DATA);
+        out_idx_dims = topKOp->get_output_partial_shape(TOPK_INDEX);
+        in_dims_size = in_dims.size();
+        out_dims_size = out_dims.size();
+
+        if (!isDynamicNgraphNode(op)) {
+            auto topKConst = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(topKOp->get_input_node_shared_ptr(TOPK_K));
+            if (!topKConst) {
+                IE_THROW() << errorPrefix <<  "gets non-constant second tensor in static shape mode!";
+            }
+            top_k = topKConst->cast_vector<int>()[0];
```

Comment:
Is there any reason why we need to read top_k inside ctor in static case? Maybe we can move this into prepareParams?

Reply:
Fixed. Thanks Dmitry!